static/app/components/sidebar/index.tsx

```diff
@@ -73,13 +76,39 @@ function Sidebar({location, organization}: Props) {
   const collapsed = !!preferences.collapsed;
   const horizontal = useMedia(`(max-width: ${theme.breakpoints.medium})`);
 
+  const hasOrganization = !!organization;
+  const [onboardingState] = usePersistedOnboardingState();
+
+  const {projects: project} = useProjects();
+
   const toggleCollapse = () => {
     const action = collapsed ? showSidebar : hideSidebar;
     action();
   };
 
   const bcl = document.body.classList;
 
+  const useOpenOnboardingSidebar = useMemo(() => {
```

Comment:
Wait this naming is weird, this isn't a hook now it's just a variable.

My suggestion was to extract this logic all out into a function called `useOpenOnboardingSidebar`

Reply:
Your hook would encapsulate all this logic, including probably also having the `useEffect` inside of it. This type of thing https://react.dev/learn/reusing-logic-with-custom-hooks